mule-extensions-api/src/main/java/org/mule/runtime/extension/api/util/ExtensionModelUtils.java

```diff
@@ -256,17 +256,19 @@ public void onSource(HasSourceModels owner, SourceModel model) {
       }
 
       private void resolve(ComponentModel model, Object owner) {
-        if (model == component) {
-          if (!(owner instanceof ExtensionModel)) {
-            result.set(owner);
-          } else if (model.requiresConnection()) {
-            extensionModel.getConfigurationModel(DEFAULT_CONFIG_NAME).ifPresent(result::set);
-          }
+        if (model == component && owner != extensionModel) {
+          result.add((ConfigurationModel) owner);
         }
       }
     }.walk(extensionModel);
 
-    return Optional.ofNullable((ConfigurationModel) result.get());
+    if (component.requiresConnection()) {
+      extensionModel.getConfigurationModel(DEFAULT_CONFIG_NAME).ifPresent(result::add);
+    }
+
+    return result.stream().sorted(reverseOrder((x, y) -> Boolean
```

Comment:
no. this method is exceeding its responsibilities. It should

a) return a list with all the matching configuration models
b) Should not filter by implicit models. That's up to the one using it. Otherwise, you're putting use case specific logic into an utility method